qcodes/station.py

```diff
@@ -684,3 +690,37 @@ def update_schema_file(
                 [qcodes.instrument_drivers] + additional_instrument_modules)
         ))
     )
+
+
+def merge_yamls(*yamls: Union[str, Path]) -> IO[str]:
+    if len(yamls) == 1:
+        return open(yamls[0], "r")
+
+    top_key = "instruments"
+    yaml = ruamel.yaml.YAML()
+
+    deq: Tdeque[Any] = deque()
+
+    # Load the yaml files and add to deque in reverse entry order
+    for filepath in yamls[::-1]:
+        with open(filepath, "r") as file_pointer:
+            deq.append(yaml.load(file_pointer))
+
+    # Add the top key entries from filepath n to filepath n-1 to ... filepath 1.
+    while len(deq) > 1:
+        data2, data1 = deq[0], deq[1]
+        for entry in data2[top_key]:
+            if entry not in data1[top_key].keys():
+                data1[top_key].update({entry: data2[top_key][entry]})
+            else:
+                raise KeyError(
+                    f"duplicate key `{entry}` detected among files: {','.join(map(str, yamls))}"
+                )
+        deq.popleft()
+
+    # Dump to a temp file so it can be read by load_config
+    merged_yaml_temp_file = tempfile.TemporaryFile("w+")
```

Comment:
Perhaps we should update this to use StringIO so we avoid writing to disk for the merge https://docs.python.org/3/library/io.html#io.StringIO